tests/integration/scale/test_scale.py

```diff
@@ -172,23 +172,30 @@ def test_scale_fail(flow_with_runtime, pod_params, mocker):
     ],
     indirect=True,
 )
-@pytest.mark.parametrize('protocol', ['websocket', 'grpc', 'http'])
-def test_scale_with_concurrent_client(flow_with_runtime, pod_params, protocol, reraise):
-    num_replicas, scale_to, _ = pod_params
-
-    def peer_client(port, protocol, peer_hash):
+@pytest.mark.parametrize('protocol', ['grpc', 'websocket', 'http'])
+def test_scale_with_concurrent_client(flow_with_runtime, pod_params, protocol):
+    def peer_client(port, protocol, peer_hash, queue):
         c = Client(protocol=protocol, port=port)
-        for _ in range(5):
-            c.index(Document(text=peer_hash), return_results=True)
+        ret = queue.get()
+        rv = c.index(Document(text=peer_hash), return_results=True)
+        for r in rv:
+            for replica_id in r.docs.get_attributes('tags__replica_id'):
+                ret.append(replica_id)
+        queue.put(ret)
+
+    num_replicas, scale_to, _ = pod_params
+    ret = []
+    queue = multiprocessing.Queue()
+    queue.put(ret)
```

Comment:
why are u putting an empty list?

Reply:
initialize the empty list to store all the replica_ids

Reply:
Review this communication pattern, u have a queue where u get and put with no control.

I think u should have the clients simply add to the queue, and assert the results from every client outside